### DISCUSSION TITLE
May I want a simpler signal connection.

### DISCUSSION THREAD

**Discussion by vinciarts** [godotengine/godot-proposals#6338] [Scripting]
I used to write Vue. The emit is very simple.
The signal and connection in GDScript are so hard to understand for me. It always drops me an error : (
I hope receiving a signal could be like this:

```
_on(signal_name, func(args))
```

**Comment by AThousandShips** [comment]
What causes an error for you?

**Comment by timothyqiu** [comment]
It's similar to `_on(signal_name, func(args))`:

```gdscript
# If you want to use `Object.connect()`, the signal should be specified as a string
connect("some_signal", some_function.bindv(args))

# But it's recommended to use the signal directly
some_signal.connect(some_function.bindv(args))

# If you don't want to pass the arguments as an array
some_signal.connect(some_function.bind(arg0, arg1, arg2))

# If you don't want to pass extra arguments
some_signal.connect(some_function)
```

Also in case you want an lambda: `some_signal.connect(func(): print("hello world"))`.

See the class reference:

* [Object.connect()](https://docs.godotengine.org/en/latest/classes/class_object.html#class-object-method-connect)
* [Signal.connect()](https://docs.godotengine.org/en/latest/classes/class_signal.html#class-signal-method-connect)
* [Callable.bindv()](https://docs.godotengine.org/en/latest/classes/class_callable.html#class-callable-method-bindv)
* [Callable.bind()](https://docs.godotengine.org/en/latest/classes/class_callable.html#class-callable-method-bind)
